packages/platform/setup-accounts.js

```diff
@@ -82,29 +92,28 @@ export default ({ mergeUserCartsOnLogin = true } = {}) => {
     check(email, Match.OneOf(String, null, undefined));
     const guestname = `${moniker.choose()}-${Random.hexString(5)}`;
     return {
-      email: email || `${guestname}@localhost`,
+      email: email || `${guestname}@unchained.local`,
       guest: true,
       profile: {},
     };
   }
 
-  Accounts.registerLoginHandler('guest', (options) => {
-    if (!options || !options.createGuest) {
-      return undefined;
-    }
-    const guestOptions = createGuestOptions(options.email);
-    return {
-      userId: Accounts.createUser(guestOptions),
-    };
-  });
+  accountsServer.services.guest = {
+    authenticate: async (params) => {
+      const guestOptions = createGuestOptions(params.email);
+      return {
+        userId: await accountsPassword.createUser(guestOptions),
+      };
+    },
+  };
 
-  Accounts.onLogin(({ methodArguments, user }) => {
+  accountsServer.on('LoginSuccess', ({ user, connection = {} }) => {
     const {
       userIdBeforeLogin,
       countryContext,
       remoteAddress,
       normalizedLocale,
-    } = [...methodArguments].pop() || {};
+    } = connection;
```

Comment:
I guess these parameters are not part of the connection, these are parameters form the GraphQL context, sure that Cart Merging works they way it should?

Reply:
>Cart Merging works they way it should
:grin: let me check.

Reply:
If not, please add a test along the way that checks if a cart is actually migrated. The idea is:

1. Login as guest
2. Put Product X in the cart
3. Login as not-guest
4. Check if logged in as the not-guest user and the cart contains Product X

Reply:
It's really important for cases where you are asked to login during the checkout